core/store/src/trie/update.rs

```diff
@@ -139,6 +155,44 @@ impl TrieUpdate {
         self.prospective.insert(trie_key.to_vec(), TrieKeyValueUpdate { trie_key, value: None });
     }
 
+    pub fn get_code(
+        &self,
+        account_id: AccountId,
+        code_hash: CryptoHash,
+    ) -> Result<Option<ContractCode>, StorageError> {
+        let key = TrieKey::ContractCode { account_id };
+        self.get(&key).map(|opt| opt.map(|code| ContractCode::new(code, Some(code_hash))))
+    }
+
+    /// Returns the size (in num bytes) of the contract code for the given account.
+    ///
+    /// This is different from `get_code` in that it does not read the code from storage.
+    /// However, the trie nodes traversed to get the code size are recorded.
+    pub fn get_code_len(
+        &self,
+        account_id: AccountId,
+        code_hash: CryptoHash,
+    ) -> Result<Option<usize>, StorageError> {
+        let key = TrieKey::ContractCode { account_id };
+        let value_ptr = self.get_ref(&key, KeyLookupMode::FlatStorage)?;
+        if let Some(value_ptr) = value_ptr {
+            let value_hash = value_ptr.value_hash();
+            debug_assert_eq!(
+                code_hash, value_hash,
+                "Hash for code does not match hash in account: Expected: {:?} Found: {:?}",
```

Comment:
nit: assert_eq should include expected and actual values in the error message, no need to manually includ eit

Reply:
done.